index.js

```diff
@@ -83,6 +99,133 @@ function attachToBody (options, req, reply, next) {
   })
 }
 
+const allowedTokens = ['[', ']', '{}']
+
+function getSerializedKeyTokens (key) {
+  const tokens = key.split(/(\[|]|{})/).filter(Boolean)
```

Comment:
We split. But why do we filter?

Reply:
This is for filtering out empty strings. It could've been written as `key.split(/(\[|]|{})/).filter(t => t)` too.